### DISCUSSION TITLE
Use parser for repl?

### DISCUSSION THREAD

**Discussion by playXE** [swc-project/swc#1482] [Q&A]
I want to add repl to [starlight](https://github.com/starlight-js/starlight) but it seems that SWC can only accept string as input for parsing, is there any way to implement REPL using swc_ecma_parser?

**Comment by kdy1** [comment]
What do you mean by `REPL`? There's alredy an unofficial repl for swc.
If you mean partial parsing, I expect it to be possible after finishing https://github.com/swc-project/swc/pull/1320.

**Reply by playXE** [reply]
I mean the one like V8 or quickjs have. Or what most web browsers have under F12 -> Console

**Reply by kdy1** [reply]
Is there a reason you can't use `String` as an input?

The parser of swc uses the [Input](https://docs.rs/swc_common/0.10.13/swc_common/input/trait.Input.html) trait of swc_common.
I can another input implementation, but I'm not sure what is the api you want.

**Reply by playXE** [reply]
Problem is not in Input trait but I believe in parser. I need API that can parse statement and return true or false when statement is finished or no. Maybe #1320 you mentioned might solve this

**Reply by kdy1** [reply]
> I need API that can parse statement and return true or false when statement is finished or no.

Then #1320 will solve the issue.
